nixos/modules/services/desktops/pipewire/pipewire-media-session.nix

```diff
@@ -0,0 +1,336 @@
+              "bluez5"
+              "restore-stream"
+              "streams-follow-default"
+            ];
+          };
+        };
+      };
+
+      alsaMonitorConfig = mkOption {
+        type = types.attrs;
+        description = ''
+          Configuration for the alsa monitor.
+        '';
+        default = {
+          # alsa-monitor config file
+          properties = {
+            #alsa.jack-device = true
+          };
+
+          rules = [
+          # an array of matches/actions to evaluate
+          {
+            # rules for matching a device or node. It is an array of
+            # properties that all need to match the regexp. If any of the
+            # matches work, the actions are executed for the object.
+            matches = [
+              {
+                # this matches all cards
+                device.name = "~alsa_card.*";
+              }
+            ];
+            actions = {
+              # actions can update properties on the matched object.
+              update-props = {
+                api.alsa.use-acp = true;
+                #api.alsa.use-ucm = true
+                #api.alsa.soft-mixer = false
+                #api.alsa.ignore-dB = false
+                #device.profile-set = "profileset-name"
+                #device.profile = "default profile name"
+                api.acp.auto-profile = false;
+                api.acp.auto-port = false;
+                #device.nick = "My Device"
+              };
+            };
+          }
+          {
+            matches = [
+              {
+                # matches all sinks
+                node.name = "~alsa_input.*";
+              }
+              {
+                # matches all sources
+                node.name = "~alsa_output.*";
+              }
+            ];
+            actions = {
+              update-props = {
+                #node.nick = 			"My Node"
+                #node.nick = 			null
+                #priority.driver = 		100
+                #priority.session = 		100
+                #node.pause-on-idle = 		false
+                #resample.quality = 		4
+                #channelmix.normalize =		false
+                #channelmix.mix-lfe = 		false
+                #audio.channels = 		2
+                #audio.format = 		"S16LE"
+                #audio.rate = 			44100
+                #audio.position = 		"FL,FR"
+                #api.alsa.period-size =         1024
+                #api.alsa.headroom =            0
+                #api.alsa.disable-mmap =        false
+                #api.alsa.disable-batch =       false
+              };
+            };
+          }
+          ];
+        };
+      };
+
+      bluezMonitorConfig = mkOption {
+        type = types.attrs;
+        description = ''
+          Configuration for the bluez5 monitor.
+        '';
+        default = {
+          # bluez-monitor config file
+          properties = {
+            # msbc is not expected to work on all headset + adapter combinations.
```

Comment:
What do we do with all the commented out configuration here? This looks like it could easily get out of date.

Reply:
I think that was what [this comment](https://github.com/NixOS/nixpkgs/pull/110615#issuecomment-769980318) and its immediate replies were getting at.

> looks like this session manager is now mandatory or it does not work

More specifically 'a' session manager is mandatory (always has been) but this is the only one that is packaged for now. Does `mkEnableOption` support setting the option to true by default?

Reply:
@jansol No, not directly but you can use lib.mkOption with type lib.bool.